frontend/src/metabase/query_builder/components/filters/BulkFilterModal/NumberFilterEditor/NumberFilterEditor.tsx

```diff
@@ -0,0 +1,132 @@
+import { useEffect } from "react";
+import { usePrevious } from "react-use";
+import { t } from "ttag";
+import _ from "underscore";
+import { Flex, NumberInput, Text } from "metabase/ui";
+import { ColumnValuesWidget } from "metabase/common/components/ColumnValuesWidget";
+import { useNumberFilter } from "metabase/common/hooks/filters/use-number-filter";
+import * as Lib from "metabase-lib";
+import type { FilterPickerWidgetProps } from "../types";
+import { FilterOperatorPicker } from "../FilterOperatorPicker";
+
+export function NumberFilterEditor({
+  query,
+  stageIndex,
+  column,
+  filter,
+  onChange,
+}: FilterPickerWidgetProps) {
+  const isID = Lib.isPrimaryKey(column) || Lib.isForeignKey(column);
+  const defaultOperator = isID ? "=" : "between";
+
+  const {
+    operator,
+    values,
+    valueCount,
+    hasMultipleValues,
+    availableOperators,
+    setOperator,
+    setValues,
+    getFilterClause,
+  } = useNumberFilter({
+    query,
+    stageIndex,
+    column,
+    filter,
+    defaultOperator,
+  });
+
+  const previousOperator = usePrevious(operator);
+  const previousValues = usePrevious(values);
```

Comment:
Something is wrong here. I believe it can be changed to avoid doing anything like that